### ISSUE TITLE
🐛 Warn criterion met, but alert state not recorded, incident not created, notification not sent

### ISSUE THREAD

**Issue by mvuets** [bosun-monitor/bosun#2188]
We've noticed odd behaviour of a Bosun alert: it did not fire up around 2017-10-27 13:36 (UTC) when it was supposed to. We presume it might be a bug in Bosun. Here is evidence I've found while digging through data.

1. If I evaluate the alert for the problematic past period, it triggers a wanted warning. So we are good here. \
![alerteval](https://user-images.githubusercontent.com/72705/32230166-d51780ac-be52-11e7-8df8-fc90bede5d51.png)
2. If I scan through the _bosun.log_ file, I see that Bosun did trigger a warning back then too. So we are good here as well.
```
2017/10/27 15:36:34 info: check.go:636: check alert XXX done (269.705663ms): 0 crits, 1 warns, 0 unevaluated, 0 unknown
```
3. What I do not see in the same log is a Bosun's attempt to send an expected notification (alike the following line). I know that under certain circumstances (e.g. already ongoing incident) notifications are not sent out again. I strongly believe it is not a case here. So we are not good on this point.
```
2017/10/31 11:50:03 info: notify.go:85: post notification successful for alert XXX. Response code 200.
```
4. The alert history page _/history?key=XXX_ does not have a record about the alert change at that moment. So we are not good here as well. \
![alerthistory](https://user-images.githubusercontent.com/72705/32230170-d87c8314-be52-11e7-9e65-0e58563556e2.png)
5. There is no corresponding incident found on the Bosun dashboard either.
6. Notifications and incidents before and after this particular case worked just fine. This is the only one that fell off and looks like a bug.

Bottom line. Bosun has recorded a warning in _bosun.log_, but has not recorded this change in the alert history (according to the corresponding web page). Incidents and notification were not fired up as well. From this perspective it looks to be an inconsistency in the Bosun's internal state which led to incorrect behaviour.

What makes this bug especially nasty is that it is impossible to spot unless we proactively watch monitored data ourselves :frowning_face: (but then why do we need Bosun? :smiley:). In other words if it repeats, we have no idea how often it happens and we effectively (and arbitrary) go blind on our monitoring.

Could you provide some insight on how we can help with troubleshooting this?

Meta:
- bosun version 0.6.0-beta1 (44426c67e5ae452f809d2dd0d7168e438afe58bc)

**Comment by digrich** [comment]
Similar behavior is noticed to me also. Bosun triggered 3 warnings but notification has been sent out for only one. This is not consistent but has been noticed a couple of times.

`2018/02/26 00:21:25 info: check.go:576: check alert LOG_ERROR done (281.108927ms): 0 crits, 3 warns, 0 unevaluated, 0 unknown`

One more strange thing, post notification message is printed twice for the incident, which is being sent out. No post notification message for other two incidents.  

`2018/02/26 00:21:26 info: notify.go:84: post notification successful for alert XXX. Response code 204.`
`2018/02/26 00:21:26 info: notify.go:84: post notification successful for alert XXX. Response code 204.`
